target/scripts/startup/setup.d/mail_state.sh

```diff
@@ -83,6 +83,9 @@ function _setup_save_states
     chown root:root /var/mail-state/spool-postfix
     # These two require the postdrop(103) group:
     chgrp -R postdrop /var/mail-state/spool-postfix/{maildrop,public}
+    # After changing the group, special bits (set-gid, sticky) are lost and we need to restore them:
+    chmod 3730 /var/spool/postfix/maildrop
+    chmod 2730 /var/spool/postfix/public
```

Comment:
These don't align with what I see when no `/var/mail-state` is present:

```console
$ exa -l --octal-permissions /var/spool/postfix

1730 drwx-wx--T - postfix  3 Mar 00:18 maildrop
0710 drwx--x--- - postfix  3 Mar 11:06 public
```

It's possible that this has changed upstream since that Aug 2018 article?

- `public/` dir only gives execute permissions for the group, and the contents there are sockets to executables, so that makes sense?
- `maildrop/` additionally allows a group to write files, with a [sticky bit](https://en.wikipedia.org/wiki/Sticky_bit) set (_[this is also a good reference](https://en.wikipedia.org/wiki/Setuid#Sticky_bit) for the various special permissions_):
  > For directories, when a directory's sticky bit is set, the filesystem treats the files in such directories in a special way so only the file's owner, the directory's owner, or [root user](https://en.wikipedia.org/wiki/Root_user) can rename or delete the file.

Increasing `maildrop/` permissions from `1730` to `3730` (_sticky bit + SGID_) doesn't seem neccessary? Nor should `public/` need to change from `0710` to `2730` (_write + SGID_).

```suggestion
    # After changing the group, special bits are lost and we need to restore them (sticky bit in this case):
    chmod 1730 /var/spool/postfix/maildrop
```

It's unlikely, but perhaps possible this was related to Postfix 2.10 changing from named pipes (FIFO) to unix sockets, but that was released in 2013, so should have been available in whatever distro the author used in 2018.

Reply:
Default permissions in Debian 11:

```
stat -c '%a %A %n' /var/spool/postfix/{maildrop,public}

1730 drwx-wx--T /var/spool/postfix/maildrop
2710 drwx--s--- /var/spool/postfix/public
```

If I stop postfix, remove those two directories and start postfix again, the directories are recreated with the following permissions:

```
stat -c '%a %A %n' /var/spool/postfix/{maildrop,public}

730 drwx-wx--- /var/spool/postfix/maildrop
710 drwx--x--- /var/spool/postfix/public
```

Reply:
I don't think I deleted the folders, forgot to mention that I did only check in an container I already had running restarted Postfix in among probably some other things, my bad! :sweat_smile: 

I have done a `git pull` and rebuilt the image, and see the same permissions @casperklein mentions without `/var/mail-state` present. I've also mounted with the state dir and exited the container to start it again with that volume still attached, permissions are untouched.

Only when removing those folders explicitly and starting Postfix did they lose those special permissions.